pipeline/extract.py

```diff
@@ -0,0 +1,37 @@
+"""This script extracts story information from the Hacker News API."""
+import pandas as pd
+from requests import get
+
+
+BASE_URL = "https://hacker-news.firebaseio.com/v0/"
+
+
+def get_top_stories() -> list:
+    """Returns the ids of the top 200 stories on Hacker News."""
+    top_stories = get(BASE_URL + "topstories.json", timeout=100).json()
+    return top_stories[:200]
+
+
+def extract_story_info(story_id: int) -> dict:
+    """Finds the details of a given story on Hacker News based on the story id."""
+    story_info = get(BASE_URL + "item/" + str(story_id) + ".json", timeout=100).json()
+    relevant_cols = ["id", "type", "by", "time", "url", "score", "title", "descendants"]
+    story_dict = {col : story_info.get(col) for col in relevant_cols}
+    return story_dict
+
+
+def main() -> None:
+    """Collects information on the 200 top stories and stores it in a csv file."""
+    all_stories = []
+    story_ids = get_top_stories()
+
+    for id in story_ids:
+        all_stories.append(extract_story_info(id))
```

Comment:
Is there any particular reason you chose not to include this loop as part of the `extract_story_info` function? 

You could rename it to `extract_all_stories_info` and use a list comprehension to return the list of all stories.

Reply:
I quite like abstracting the function to have a more basic functionality, but I'm going to update it to be a list comprehension within the main function

Reply:
I agree, it's simpler like this. Both are fine!